src/plugins/passwd/README.md

```diff
@@ -15,7 +15,16 @@ This plugin parses `passwd` files, e.g. `/etc/passwd`.
 
 ## Implementation Details
 
-If present, the not-posix compliant `fgetpwent` function will be used to read the file supplied by the resolver, otherwise `getpwent` will be used. For writing, if present, `putpwent` will be used, if not a simple implementation writing straight to the config file.
+If present, the not-posix function `fgetpwent` (GNU_SOURCE) will be used to
+read the file supplied by the resolver.
+Alternatively `getpwent` (X_OPEN_SOURCE >= 500) will be used.
+
+For writing, if present, `putpwent` (DEFAULT_SOURCE) will be used, if not a
```

Comment:
This sentence seems to be a little complicated. Maybe you could use two sentences instead of one:

> For writing, if present, `putpwent` (DEFAULT_SOURCE) will be used. If `putpwent` is not available, the plugin will write straight to the config file.

.